drivers/CpuUid.cpp

```diff
@@ -0,0 +1,80 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "hal/cpu_uid_api.h"
+#include "platform/mbed_assert.h"
+#include "drivers/CpuUid.h"
+
+#if DEVICE_CPUUID
+
+namespace mbed {
+
+const char CpuUid::_hexChars[16] = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F'};
+
+CpuUid::CpuUid() : _data(NULL)
+{
+    _size = cpu_uid_get_length();
+#ifdef MBED_ASSERT
```

Comment:
Why does the assert need to be wrapped? We have build profiles for this.

Reply:
There were build problems on some targets, and I wanted to get rid of them ... then I realized that I should include mbed_assert.h to get rid of build errors ... I just forgot to remove the ifdef - will fix it soon ;)